index.html

```diff
@@ -296,141 +292,170 @@
+      var image_name = scheduleObj.name.toLowerCase();
+      image_name = image_name.replace("/", "");
+      image_name = image_name.replace(/\s/g, '');
+      scheduleObj.avatar = "/images/" + image_name + ".jpg";
 
-        } else if (type == "lite") {
-          if (letterToNum(scheduleObj.period) <= 8) { // If it's a class period
-            scheduleObj.shared = false; // Find out if the user has a class that period
-            for (var i = 0; i < userSchedule.classes.length; i++) {
-              if (userSchedule.classes[i].period == scheduleObj.period &&
-                  userSchedule.classes[i].name == scheduleObj.name) {
-                scheduleObj.shared = true;
-                break;
-              }
-            }
-          } else {
-            continue; // Skip over lunch, advisory, assembly, etc.
-          }
-        }
+    // If the class is a normal class but the user is a teacher
+    } else {
+      sanitized_teacher_name = userSchedule.firstname + "_" + userSchedule.lastname;
+      sanitized_teacher_name = sanitized_teacher_name.toLowerCase();
+      scheduleObj.avatar = "/96x96_photos/" + sanitized_teacher_name + ".jpg";
+    }
+    scheduleObj.roomLink = "/room/" + scheduleObj.room.toLowerCase().replace(/ /g,"_");
+    return scheduleObj;
+  }
 
-        var times = days[0][day][school][currentSlot]["time"].split('-');
-        scheduleObj.startTime = times[0].trim();
-        scheduleObj.endTime = times[1].trim();
-        todaySchedule.push(scheduleObj);
+  function isSharedClass(scheduleObj) {
+    for (var i = 0; i < userSchedule.classes.length; i++) {
+      if (userSchedule.classes[i].period == scheduleObj.period &&
+          userSchedule.classes[i].name == scheduleObj.name) {
+        return true;
+      }
     }
+    return false;
+  }
 
-    scheduleElement.entries = todaySchedule;
-    if (todaySchedule.length == 0) { // If there are no classes
-      var imageNum = Math.floor((Math.random() * NUM_NOSCHOOL_IMAGES) + 1);
-      var imageUrl = "/images/noschool" + imageNum.toString() + ".jpg";
-      var imageText = "No School";
-      scheduleElement.imageProperties = {url: imageUrl, text: imageText};
-    } else {
-      scheduleElement.imageProperties = null;
+  function renderSchedule(dateObj, schedule, type, scheduleElement) {
+    // Either MS or US
+    var school = getSchool(schedule.grade);
+    var todaySchedule = [];
+    // The current schedule for that day
+    var day = getScheduleTypeForDate(dateObj, days);
+    // If it is a normal day (no all day event)
+    if (!days[0][day].allDayEvent) {
+      // For each class
+      for (var currentSlot = 0; currentSlot < days[0][day][school].length; currentSlot++) {
+        // Render the base of the class
+        var scheduleObj = createClassEntry(schedule, school, day, currentSlot, type);
+        if (scheduleObj) { // If scheduleObj is not null
+          // To see when scheduleObj might be null, look at createClassEntry()
```

Comment:
Yes, if it is a lite schedule and the class is one of the non-core classes (e.g. advisory).